### DISCUSSION TITLE
Reindent file

### DISCUSSION THREAD

**Discussion by NickNick** [helix-editor/helix#9869] [Q&A] [answered]
I'd like to reindent a whole file, but can't figure out how to. What I mean by reindenting is that, when I'm editing this file and hit enter somewhere, the cursor jumps to the right depth (for the current indentation settings). I'd like to do that for the whole file.

`format` does not work because I don't have an LSP for this file type.

**Comment by kirawi** [comment, marked as answer]
This isn't possible and I'm not aware of a tracking issue for this. I _think_ Vim supports this so it might be worth considering.

**Reply by trev-dev** [reply]
Both Emacs and Vim have a fallback, rudimentary method to reindent a line, selection and file.

**Reply by ronisbr** [reply]
At least neovim has a fallback based on tree-sitter. This can be really good. For many cases, I turn off the LSP formatting and use this one.
